core/src/main/java/com/graphhopper/routing/util/WheelchairFlagEncoder.java

```diff
@@ -0,0 +1,286 @@
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.util;
+
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.storage.IntsRef;
+import com.graphhopper.util.EdgeIteratorState;
+import com.graphhopper.util.PMap;
+import com.graphhopper.util.PointList;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeMap;
+
+import static com.graphhopper.routing.util.PriorityCode.REACH_DEST;
+import static com.graphhopper.routing.util.PriorityCode.VERY_NICE;
+
+/**
+ * A flag encoder for wheelchairs.
+ *
+ * @author don-philipe
+ */
+public class WheelchairFlagEncoder extends FootFlagEncoder {
+    private final Set<String> excludeSurfaces = new HashSet<>();
+    private final Set<String> excludeSmoothness = new HashSet<>();
+    private final Set<String> excludeHighwayTags = new HashSet<>();
+    private final int maxInclinePercent = 6;
+
+    /**
+     * Should be only instantiated via EncodingManager
+     */
+    public WheelchairFlagEncoder() {
+        this(4, 1);
+    }
+
+    public WheelchairFlagEncoder(PMap properties) {
+        this((int) properties.getLong("speed_bits", 4),
+                properties.getDouble("speed_factor", 1));
+        this.setBlockFords(properties.getBool("block_fords", true));
+    }
+
+    public WheelchairFlagEncoder(int speedBits, double speedFactor) {
+        super(speedBits, speedFactor);
+        restrictions.add("wheelchair");
+
+        setBlockByDefault(false);
+        absoluteBarriers.add("handrail");
+        absoluteBarriers.add("wall");
+        absoluteBarriers.add("turnstile");
+        potentialBarriers.add("kerb");
+        potentialBarriers.add("cattle_grid");
+        potentialBarriers.add("motorcycle_barrier");
+
+        safeHighwayTags.add("footway");
+        safeHighwayTags.add("pedestrian");
+        safeHighwayTags.add("living_street");
+        safeHighwayTags.add("residential");
+        safeHighwayTags.add("service");
+        safeHighwayTags.add("platform");
+
+        excludeHighwayTags.add("trunk");
+        excludeHighwayTags.add("trunk_link");
+        excludeHighwayTags.add("primary");
+        excludeHighwayTags.add("primary_link");
+        excludeHighwayTags.add("secondary");
+        excludeHighwayTags.add("secondary_link");
+        excludeHighwayTags.add("tertiary");
```

Comment:
Is tertiary a Kreisstraße/Landstraße? Then please don't exclude. Sometimes, that's the only available way in rural areas - except for forrest/agricultural roads (Feldwege), wich are much worse.